### PR TITLE
Add resource list filter control

### PR DESCRIPTION
**Pull request by andrewpye**
Adds an implementation of the React [`ResourceList/FilterControl`](https://github.com/Shopify/polaris-react/tree/master/src/components/ResourceList/components/FilterControl) component.

### CODE REVIEW

File: addon/components/polaris-resource-list/filter-control.js [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+   * for rendering in the template
+   *
+   * @property appliedFiltersForRender
+   * @type {Object[]}
+   * @private
+   */
+  appliedFiltersForRender: computed('appliedFilters.[]', function() {
+    let appliedFilters = this.get('appliedFilters') || [];
+    return appliedFilters.map((appliedFilter) => {
+      let appliedFilterForRender = JSON.parse(JSON.stringify(appliedFilter));
```
Comment (tomnez) [inline addon/components/polaris-resource-list/filter-control.js]: I wonder if there will ever be a nice, normal way to copy objects in JavaScript 🙏

File: addon/components/polaris-resource-list/filter-control.js [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+  return new Date(date.replace(/-/g, '/')).toLocaleDateString();
+}
+
+function findOperatorLabel(filter, appliedFilter) {
+  let operatorText = get(filter, 'operatorText');
+
+  if (
+    filter.type === FilterType.DateSelector &&
+    (get(appliedFilter, 'key') === get(filter, 'minKey') ||
+      get(appliedFilter, 'key') === get(filter, 'maxKey'))
```
Comment (tomnez) [inline addon/components/polaris-resource-list/filter-control.js]: It seems you have also been bitten by the weird `||` indentation bug 😝

File: addon/components/polaris-resource-list/filter-control.js [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+    );
+  }
+}
+
+/**
+ * This function is a workaround for the fact that the
+ * React implementation uses the Polaris i18n service
+ * internally, which we don't have.
+ */
+function getFilterLabelForValue(value) {
```
Comment (tomnez) [inline addon/components/polaris-resource-list/filter-control.js]: 👍

File: addon/components/polaris-resource-list/filter-control.js (hunk 1 above)
Comment (andrewpye) [reply addon/components/polaris-resource-list/filter-control.js]: We can only hope 🙏 😛

File: addon/components/polaris-resource-list/filter-control.js (hunk 2 above)
Comment (andrewpye) [reply addon/components/polaris-resource-list/filter-control.js]: But it's _prettier_ this way :trollface:

**Pull request merged**